packages/vite/src/node/utils.ts

```diff
@@ -1222,6 +1234,10 @@ export function joinUrlSegments(a: string, b: string): string {
   return a + b
 }
 
+export function initialSlashRemoved(str: string): string {
+  return str.startsWith('/') ? str.slice(1) : str
```

Comment:
`str[0] === '/'` ?

<img width="959" alt="image" src="https://user-images.githubusercontent.com/102238922/226791333-6a8d7e21-e39b-4221-9b6d-9a39b2699d18.png">

Reply:
That's interesting, I think we used `startsWith` in quite a few other places too (and I usually use that). Do you have a link to the benchmark?

Reply:
Quite a big diff... changed here. @sun0day maybe you could change this in other places in another PR if you would like.

Reply:
https://perf.link/

I saw sapphi-red mentioned this link in another pr.

Reply:
The URL changes when you do a test in perf.link so you can share that directly instead (or with) the screenshot so others can continue playing and running it on their machines too.

Reply:
[`startsWith` source](https://github.com/zloirock/core-js/blob/446aa802ebc37831c14c5f18be17d01189374bd5/packages/core-js/modules/es.string.starts-with.js#L26-L36)

[perf test](https://perf.link/#eyJpZCI6InFmNmVhMmwzMTNnIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSAnYWFhJyIsInRlc3RzIjpbeyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6ImRhdGEuc3RhcnRzV2l0aCgnYicpIiwicnVucyI6WzM5MzAwMCwxNDI2MDAwLDkwMDAsMjQ1NjAwMCwzNjUyMDAwLDQ1MDQwMDAsNDgzNjAwMCwxODA4MDAwLDMwNDEwMDAsMjE5MjAwMCw0NTU4MDAwLDIyMjgwMDAsMjY0NzAwMCw0MzEwMDAwLDI1MzcwMDAsMTg5NTAwMCw1Mzc5MDAwLDQ5NTMwMDAsMzMyNjAwMCw1MjE5MDAwLDQ4NzIwMDAsMzEwMzAwMCwxNjQ2MDAwLDQwMzcwMDAsNDYwMDAwMCwzNjUxMDAwLDI1MTkwMDAsMTM2OTAwMCw1MjM1MDAwLDIzNTAwMDAsNDc1NDAwMCw1NDI4MDAwLDI3MDAwMDAsNDc0MjAwMCw4MzgzMDAwLDQyMDQwMDAsNDYwMjAwMCwyNjczMDAwLDU1NjAwMCwyNzUyMDAwLDIxNDUwMDAsNTM1MzAwMCw0NDExMDAwLDM1NDAwMDAsNTA0MjAwMCw1NDI4MDAwLDIxODQwMDAsNDA4MjAwMCw1ODcwMDAsMzMxOTAwMCw1MzgxMDAwLDQ1MDcwMDAsMzIwMzAwMCwxODQzMDAwLDIxMTYwMDAsMjkxNjAwMCwyMTM4MDAwLDIxMjAwMDAsMzM2ODAwMCw0MjM4MDAwLDYzMDAwMCwxNDAyMDAwLDQ2MTcwMDAsNjk0MDAwLDQ4MDUwMDAsNjY2OTAwMCwyNjg3MDAwLDI2MjkwMDAsMzcyNzAwMCw1MDE5MDAwLDMzMDIwMDAsNTE1MzAwMCw0MjYwMDAwLDgxNzUwMDAsNDU4OTAwMCw2ODA5MDAwLDQxMjUwMDAsMzAzNDAwMCw1MjEzMDAwLDQ5MjYwMDAsMTQwNTAwMCwzODc0MDAwLDE4NzAwMDAsMzc2MjAwMCwzOTY1MDAwLDM4MzAwMCw0NTUxMDAwLDMzNjcwMDAsMjY3NDAwMCwyOTg3MDAwLDQzODcwMDAsMzgxMjAwMCw1MTEyMDAwLDY3NjcwMDAsMzQyODAwMCwyNjk3MDAwLDQxMTYwMDAsNDI0NzAwMCw3NjE2MDAwLDE1MzEwMDBdLCJvcHMiOjM1ODQ4MjB9LHsibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJkYXRhWzBdID09PSAnYiciLCJydW5zIjpbMTM3MzAwMCw0NDY4MDAwLDMxNDIwMDAsMTA5NDAwMCwyNDc3MDAwLDMxMjgwMDAsMTQ1ODAwMCw0OTAzMDAwLDg1NTEwMDAsMjA0MDAwMCwxMzMzMDAwLDg5NDAwMCw1NzczMDAwLDIyMTYwMDAsODYyMDAwLDkxODkwMDAsMjE4NTAwMCw2MTE2MDAwLDc3MDcwMDAsMzI0NTAwMCwzOTg0MDAwLDM2ODAwMCw3NDQ0MDAwLDcwMTAwMCw3NzI2MDAwLDExMTMwMDAsODk2ODAwMCw5NDc1MDAwLDMwMzUwMDAsNzgwNTAwMCwzNjIyMDAwLDg0NTgwMDAsODk3MjAwMCw0MjQ0MDAwLDUxODEwMDAsOTAxOTAwMCwyODA5MDAwLDg1MjAwMDAsODc0MDAwLDQ3MDAwLDkxOTMwMDAsNjQwNTAwMCw0MTQ2MDAwLDIxNDgwMDAsMTA3MjAwMCw0MDQ2MDAwLDg1NTgwMDAsMTc0ODAwMCwyMjYzMDAwLDI3NzQwMDAsMjUyNzAwMCwxMzMwMDAsOTM0NjAwMCwyMjUwMDAsODEyMDAwMCw1NzczMDAwLDg0NzIwMDAsMzQxNTAwMCw4MjIwMDAwLDIxNzIwMDAsMjcyMDAwLDc1NzEwMDAsNjIzOTAwMCw5NzQ4MDAwLDIxNjkwMDAsMzMyNTAwMCw5NzAwMCw0MDAwMDAsMzc2MDAwLDMwOTcwMDAsNDcwMDAsMTgzOTAwMCwxOTQ3MDAwLDQ2MzkwMDAsMTUzODAwMCwzNTMwMDAwLDMyOTcwMDAsNjIzNDAwMCw0MDc5MDAwLDE4MzkwMDAsODYwMzAwMCwyMjMxMDAwLDI4MTAwMCwxNTc4MDAwLDEwNjEwMDAsODcwNjAwMCwyMzA4MDAwLDE1NjQwMDAsNzA1MDAwLDk1NjEwMDAsMjMzMzAwMCwyMDk5MDAwLDE4NTkwMDAsNTIxNDAwMCw2MzgwMDAsOTIxNDAwMCwxNjg0MDAwLDgwMzAwMCw1OTgxMDAwLDg0MzIwMDBdLCJvcHMiOjQwMjQxMzB9XSwidXBkYXRlZCI6IjIwMjMtMDMtMjJUMDY6MzA6MzUuMDQ1WiJ9)

Reply:
LOVE the site! Let's go with `[0] === '/'` then. It would be good to change this in other places like patak mentioned.